hoomd/hpmc/pytest/conftest.py

```diff
@@ -0,0 +1,1271 @@
+                     {'vertices': [(1, 0, 0), (1, 1, 0), (1, 2, 1),
+                                   (0, 1, 1), (1, 1, 2), (0, 0, 1)],
+                      'ignore_statistics': 0,
+                      'sweep_radius': 1.0},
+                     {'vertices': [(0, 0, 0), (1, 1, 1), (1, 0, 2),
+                                   (2, 1, 1)],
+                      'ignore_statistics': 1,
+                      'sweep_radius': 0}]
+        return args_list
+    return get_args
+
+
+@pytest.fixture(scope="function")
+def convex_polyhedron_invalid_args():
+    def get_args():
+        args_list = [{'vertices': "invalid",
+                      'ignore_statistics': 1,
+                      'sweep_radius': 2.0},
+                     {'vertices': 1,
+                      'ignore_statistics': 0,
+                      'sweep_radius': 1.0},
+                     {'vertices': [(0, 0, 0), (1, 1, 1), (1, 0, 2),
+                                   (2, 1, 1)],
+                      'ignore_statistics': 1,
+                      'sweep_radius': "invalid"},
+                     {'vertices': [(0, 0, 0), (1, 1, 1), (1, 0, 2),
+                                   (2, 1, 1)],
+                      'ignore_statistics': "invalid",
+                      'sweep_radius': "invalid"}]
+        return args_list
+    return get_args
+
+
+@pytest.fixture(scope="function")
+def convex_spheropolygon_integrator():
+    def make_shape():
+        return hoomd.hpmc.integrate.ConvexSpheropolygon
+    return make_shape
+
+
+@pytest.fixture(scope="function")
+def convex_spheropolygon_valid_args(convex_polygon_valid_args):
+    def get_args():
+        return convex_polygon_valid_args()
+    return get_args
+
+
+@pytest.fixture(scope="function")
+def convex_spheropolygon_invalid_args(convex_polygon_invalid_args):
+    def get_args():
+        return convex_polygon_invalid_args()
+    return get_args
+
+
+@pytest.fixture(scope="function")
+def convex_spheropolyhedron_integrator():
+    def make_shape():
+        return hoomd.hpmc.integrate.ConvexSpheropolyhedron
+    return make_shape
+
+
+@pytest.fixture(scope="function")
+def convex_spheropolyhedron_valid_args(convex_polyhedron_valid_args):
+    def get_args():
+        return convex_polyhedron_valid_args()
+    return get_args
+
+
+@pytest.fixture(scope="function")
+def convex_spheropolyhedron_invalid_args(convex_polyhedron_invalid_args):
+    def get_args():
+        return convex_polyhedron_invalid_args()
+    return get_args
+
+
+@pytest.fixture(scope="function")
+def ellipsoid_integrator():
+    def make_shape():
+        return hoomd.hpmc.integrate.Ellipsoid
+    return make_shape
+
+
+@pytest.fixture(scope="function")
+def ellipsoid_parameters():
+    def get_parameters():
+        return hoomd.hpmc._hpmc.EllipsoidParams
+    return get_parameters
+
+
+@pytest.fixture(scope="function")
+def ellipsoid_valid_args():
+    def get_args():
+        args_list = [{"a": 0.75, "b": 1, "c": 0.5, 'ignore_statistics': 0},
```

Comment:
Could we have optional parameters not set in one of the options.

Reply:
I am getting a KeyError when initializing all of the C++ integrators if I don't explicitly set each parameter, see d83f466fe

Reply:
Made separate fixture for calling c++ objects directly 13e7126f9